maven/build-scan-save/action.yml

```diff
@@ -0,0 +1,16 @@
+name: Save Maven Build Scan
+description: Save Maven Build Scan
+
+runs:
+  using: composite
+  steps:
+    - name: Generate UUID
+      id: generate-uuid
+      run: |
+        echo "UUID=$(cat /proc/sys/kernel/random/uuid)" >> "$GITHUB_OUTPUT"
+      shell: bash
+    - name: Upload Build Scan as workflow Artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: 'scan-maven-${{ steps.generate-uuid.outputs.UUID }}'
```

Comment:
Why is the UUID required?

My understanding is that it's [OK to upload multiple times with the same artifact](https://github.com/actions/upload-artifact#uploading-to-the-same-artifact), as long as the underlying files have different names. (Which they definitely do in case of Gradle `build-scan-data`, not sure about Maven).

That way, we'll have a single `scans-maven` artifact to restore in the next story, rather than a series of prefixed artifacts.

Reply:
I think I misinterpreted the doc, to me 2 problems could occur:
- the artifact being corrupted when uploaded to at the same time
- an upload failing in HTTP 503 when uploading same file/path
Re-reading I feel you have the right interpretation.

The underlying file is always scan.scan but belongs to a different folder with an apparently unique name for Gradle and Maven.
Having one single artifact will greatly simplify.

Reply:
I removed the UUID 👍

Reply:
Excellent!